cli/EAP7-1100-Add_the_ability_for_the_CLI_SSL_security_commands_to_be_able_to_obtain_a_server_certificate_from_Lets_Encrypt.adoc

```diff
@@ -0,0 +1,114 @@
+= EAP7-1100 Add the ability for the CLI SSL security commands to be able to obtain a server certificate from Let's Encrypt
+:author:            Marek Marusic
+:email:             mmarusic@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          cli,add,Let's,Encrypt,Let's Encrypt,security,commands
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+WFCORE-3447 introduced high-level security commands (ssl enable-ssl-management and ssl enable-ssl-http-server)
+for enabling one-way and two-way SSL easily.
+To generate the server certificate,
+these high-level security commands made use of key-store management operations to generate a self-signed certificate.
+Since it is now possible to obtain certificates from the Let's Encrypt certificate authority using the CLI,
+the high-level security commands should be updated so that in addition to being able to generate a self-signed server certificate,
+they can also obtain a server certificate from Let's Encrypt.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1100[EAP7-1100]
+* https://issues.jboss.org/browse/WFCORE-4227[WFCORE-4227]
+
+=== Related Issues
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:jdenise@redhat.com[Jean-François Denise]
+
+=== QE Contacts
+
+* mailto:mchoma@redhat.com[Martin Choma]
+
+=== Affected Projects or Components
+
+* CLI
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+* Add parameter '--lets-encrypt' to the 'enable-ssl-http-server' and 'enable-ssl-management' security commands
+* The parameter '--lets-encrypt' should be used in combination with the '--interactive' parameter
+* After '--lets-encrypt parameter' is used user is asked for several information needed to obtain the certificate:
+** Key-store file name
+** Password
+** Domain name(s)
+** Contact url(s)
+** Validity
+** Alias
+** Enable SSL Mutual Authentication
+* Introduce '--ca-account=<name>' parameter to reuse existing --ca-account
+
```

Comment:
I think this should also take a boolean parameter to indicate whether or not the user agrees to Let's Encrypt's terms of service. (The ```obtain-certificate``` operation takes ```agree-to-terms-of-service``` as an optional parameter. It only needs to be specified the first time the user obtains a certificate from Let’s Encrypt with their ```certificate-authority-account```.)